src/common/type/stringId.h

```diff
@@ -0,0 +1,123 @@
+#ifndef COMMON_TYPE_STRINGID_H
+#define COMMON_TYPE_STRINGID_H
+
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "common/type/string.h"
+
+/***********************************************************************************************************************************
+Maximum number of characters in a StringId. This is a safe buffer size when calling strIdToZN. If the buffer needs to be
+zero-terminated then an extra byte should be added.
+***********************************************************************************************************************************/
+// Maximum for specific encodings (e.g. 5-bit, 6-bit)
+#define STRID5_MAX                                                  13
+#define STRID6_MAX                                                  11
+
+// Maximum for any encoding
+#define STRID_MAX                                                   STRID5_MAX
+
+/***********************************************************************************************************************************
+StringId typedef to make them more recognizable in the code
+***********************************************************************************************************************************/
+typedef uint64_t StringId;
+
+/***********************************************************************************************************************************
+Number of bits to use for encoding. The number of bits affects the character set that can be encoded.
+***********************************************************************************************************************************/
+typedef enum
+{
+    stringIdBit5 = 0,                                               // 5-bit encoding for a-z, 1-4, and - characters
+    stringIdBit6 = 1,                                               // 6-bit encoding for a-z, 0-9, A-Z, and - characters
+} StringIdBit;
+
+/***********************************************************************************************************************************
+Macros to define constant StringIds. ALWAYS use strIdGenerate() to create these macros. The parameters in the macros are not
+verified against each other so the str parameter is included only for documentation purposes.
+***********************************************************************************************************************************/
+#define STRID5(str, strId)                                          strId
+#define STRID6(str, strId)                                          strId
+
+/***********************************************************************************************************************************
+Functions
+***********************************************************************************************************************************/
+// Convert N chars to StringId, If the string is longer than the allowable length for the selected encoding then the StringID will
```

Comment:
In this statement, I took `allowable length` to be the STRID5_MAX, or STRID6_MAX, but a string that is 11 characters `strIdGenerate("mytest0123a")` - is equal to, not longer than, the allowable length of 11 yet when I convert it back, the 11th character becomes a + sign - so is there something I'm missing?

Reply:
The max includes the + indicator when present. I clarified the comment a bit in aabc411388a3ca.